ouroboros-consensus/src/Ouroboros/Consensus/Mempool/TxSeq.hs

```diff
@@ -53,7 +60,14 @@ data TxTicket tx = TxTicket !tx !TicketNo
 -- splitting and indexing by the ticket number.
 --
 newtype TxSeq tx = TxSeq (StrictFingerTree TxSeqMeasure (TxTicket tx))
-  deriving Show
+  deriving (Show)
+
+instance NoUnexpectedThunks tx => NoUnexpectedThunks (TxSeq tx) where
+  showTypeOf _ = "TxSeq"
+  whnfNoUnexpectedThunks ctxt (TxSeq ft) =
+      -- We cannot rely on our own 'Foldable' instance here because it
+      -- /introduces/ thunks
+      noUnexpectedThunksInValues ctxt (Foldable.toList ft)
```

Comment:
Is there no instance for `StrictFingerTree`?

Reply:
There is now..

Reply:
Or rather, we could make one now.

Reply:
Done.